### DISCUSSION TITLE
v0.13.0 Testing Checklist

### DISCUSSION THREAD

**Discussion by jorgeepc** [kubeshop/tracetest#2958] [Release testing]
# Testing Tracetest

This is the entire checklist on what we should do to assert that Tracetest is working fine on each version release. On each version release, we can copy the contents of this template checklist and open a Github Discussion to start each test.

On Tracetest, we work in two ways to test the entire system and guarantee that everything is working fine. One is through automatic tests, where we run unit, integration and end-to-end tests on Tracetest CLI, Web UI and Server.

Another source of tests is the manual tests that we execute on each release, simulating users, and checking if everything is ok on these tests.

## Manual Tests

On our manual tests, we aim to do some [sanity checks](https://en.wikipedia.org/wiki/Sanity_check) to see if the main features are working as expected. Usually, we run it on each Tracetest release.

### Tests to validate on Release Candidate

- Test server installation via CLI
  - [x] Docker Compose and no demo API
  - [x] Docker Compose and demo API
  - [x] Kubernetes and no demo API
  - [x] Kubernetes and demo API
  - [x] Verify that Tracetest is not sending analytics if you have the `TRACETEST_DEV` env for docker install
  - [x] Verify that Tracetest is not sending analytics if you have the `TRACETEST_DEV` env for k8s install

- Test DataStore provisioning and basic test
  - [x] Open WebUI and go to `/settings` page. The provisioned Data Store should be selected
  - [x] Create a test on WebUI that calls a demo API (like [Pokeshop](https://docs.tracetest.io/live-examples/pokeshop/overview) or [Open Telemetry Store](https://docs.tracetest.io/live-examples/opentelemetry-store/overview)). This test should fetch traces correctly and run without errors

- Test Tracetest examples
  - [x] [Jaeger example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-jaeger)
  - [x] [Tempo example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-tempo)
  - [x] [New Relic example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-new-relic)
  - [x] [Lightstep example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-lightstep)
  - [x] [OpenSearch example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-opensearch)
  - [x] [Elastic APM example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-elasticapm-with-elastic-agent)
  - [x] [SignalFX example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-signalfx)
  - [x] [Datadog example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-datadog)
  - [x] [Amazon X-Ray example](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-amazon-x-ray)
  - [x] [Honeycomb](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-honeycomb)
  - [x] [Azure Application Insights](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-azure-app-insights)

- Test CLI commands (detailed list [here](https://github.com/kubeshop/tracetest/tree/main/testing/cli-e2etest#cli-commands-to-test))

  - [x] Resources: config
  - [x] Resources: datastore
  - [x] Resources: demo
  - [x] Resources: environment
  - [x] Resources: pollingprofile
  - [x] Resources: analyzer
  - [x] Resources: test
  - [x] Resources: transaction
  - [x] Test Runner

- Testing benchmark
  - [x] Run OpenTelemetry Demo tests with the current version and guarantee that the CLI and Server responded correctly

- Test specific features added/changed on this release:
  - [x] CLI run command for tests (by file, by ID) #2885
  - [x] CLI run command for transactions (by file, by ID) #2885
  - [x] Web automate page CLI for tests #2928
  - [x] Web automate page CLI for transactions #2939
  - [x] Docker host URL warning #2934
  - [x] CLI TestRunner resource #2951
  - [x] Web TestRunner resource #2922
  - [x] CLI test run with required gates #2944
  - [x] Web test run with required gates #2922 
  - [x] Signoz as a Datastore #2935
  - [x] Validate environment search #2956 
  - [x] Validate test search #2957 
  - [x] Span Events with assertions for Jaeger #2911
  - [x] Span Events with assertions for Tempo #2911
  - [x] Span Events with assertions for OTLP #2911
  - [x] Span Events with assertions for App Insights #2911
  
### Tests to validate on Final Release

- Pipeline
  - [x] Check if our release pipeline on [Release Tracetest](https://github.com/kubeshop/tracetest/actions/workflows/release-version.yml) workflow on Github Actions worked correctly

- Test CLI updates
  - [x] MacOS via homebrew
  - [x] MacOS via curl script
  - [ ] Windows via chocolatey
  - [ ] Windows via manual download

- Docs
  - [x] Double check [Detailed installation](https://docs.tracetest.io/getting-started/installation) docs and see if everything is documented correctly

- Add Major Changes
  - [x] Add a summary with major changes to the release page

- Test specific features added/changed on this release:
  - [ ] Add list of features here

**Comment by xoscar** [comment]
Bugs:
- [ ] [[Backend] Jaeger events are missing the name](https://github.com/kubeshop/tracetest/issues/2961)
- [ ] [[Backend] Azure App Insights Pokeshop integration has missing events](https://github.com/kubeshop/tracetest/issues/2962)
- [ ] [[Frontend] Performance issue when rendering large responses #2963](https://github.com/kubeshop/tracetest/issues/2963)
